packages/perspective/test/js/to_format.spec.js

```diff
@@ -431,8 +431,8 @@ const pivoted_output = [
             let view = await table.view();
             let json = await view.to_json({ formatted: true });
             expect(json).toEqual([
-                { datetime: "6/13/16" },
-                { datetime: "6/14/16" },
+                { datetime: "2016-06-13" },
+                { datetime: "2016-06-14" },
```

Comment:
Curious, how important is it to maintain datatime strring parsing across different systems? Since it's possible to use a parsed date string as part of a column name, should we at one point add a C++ based function that both JS and Python can call to get a standardized formatted date string?

Reply:
There are only two conditions where `date`/`datetime` columns need to be formatted:

(1) When they are used in a `split_by` and we need to stringify them to create compound column names. This is really a deficiency of the JSON encoding we use which doesn't permit true column paths. Fixing this encoding would remove the need for formatting for this case.

(2) When the `formatted` flag is passed. This is only used by the CSV exporter (as users typically expect CSVs to be "human readable"). Formerly, this used `sprintf` to try to replicate the `en/us` locale, as the browser native locale is not cheaply accessible from C++; however, this was never right, especially in non-`en/us` locales. The format in this patch is a compromise - it is not locale-formatted, but it is consistently emitted and parsed (though assumption are made about the encoding timezone in this case which are symmetric wrt Perspective but maybe not always desired).